### PR TITLE
Features/rhpm cd02

### PR DESCRIPTION
**Pull request by karthikkharvi**
Uploading documents of candidate

### CODE REVIEW

File: api/RS.API/RS.Web/Controllers/CandidateController.cs [hunk 1]
```diff
@@ -82,5 +102,47 @@ public IResult ApprovedForInterview(Guid candidateId)
             var approvedCandidate = _candidateManagerService.ApprovedForInterview(candidateId);
             return approvedCandidate;
         }
+
+        private string GetExtension(IFormFile file)
```
Comment (salilsaini868) [inline api/RS.API/RS.Web/Controllers/CandidateController.cs]: create common methods for file handling under common layer.

File: api/RS.API/RS.Service/Logic/CandidateManagerService.cs [hunk 2]
```diff
@@ -315,31 +343,50 @@ public IResult UpdateCandidate(CandidateViewModel candidate)
...
-                    updatedOpeningCandidate.OpeningId = candidate.Opening;
-                    updatedOpeningCandidate.MapAuditColumns((ClaimsIdentity)_principal.Identity);
-                }
-                _candidateRepository.UpdateCandidate(candidateModel, updatedOpeningCandidate, organization);
-                result.Body = candidate.CandidateId;
+                    candidateDocumentModel.MapFromViewModel(candidateDocumentViewModel, (ClaimsIdentity)_principal.Identity);
+                } 
+                #endregion
+
+                _candidateRepository.UpdateCandidate(candidateModel, openingCandidateModel, candidateDocumentModel);
```
Comment (salilsaini868) [inline api/RS.API/RS.Service/Logic/CandidateManagerService.cs]: attach candidatedocument model to candidate model as they have navigation

File: api/RS.API/RS.Web/Controllers/CandidateController.cs [hunk 3]
```diff
@@ -82,5 +102,47 @@ public IResult ApprovedForInterview(Guid candidateId)
...
+        }
+
+        private void SaveFile(IFormFile file)
+        {
+            string path = Path.Combine(_hostingEnvironment.ContentRootPath, "uploadFiles");
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+            var ext = new List<string> { ".pdf", ".doc", ".docx" };
```
Comment (salilsaini868) [inline api/RS.API/RS.Web/Controllers/CandidateController.cs]: move extension under api project .json file, in future these extension might be increased or decreased

**Review by salilsaini868 — CHANGES_REQUESTED** [review]

File: api/RS.API/RS.Web/Controllers/CandidateController.cs (hunk 3 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Web/Controllers/CandidateController.cs]: Done

File: api/RS.API/RS.Service/Logic/CandidateManagerService.cs (hunk 2 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Service/Logic/CandidateManagerService.cs]: Done

File: api/RS.API/RS.Web/Controllers/CandidateController.cs (hunk 1 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Web/Controllers/CandidateController.cs]: Done

**Review by salilsaini868 — APPROVED** [review]

**Pull request merged**